# Hyperparameter Tuning

This notebook summarizes the hyperparameter tuning results for the best-performing classical machine learning models.

In [1]:
from pathlib import Path
import pandas as pd

PROJECT_ROOT = Path("..")
REPORTS_DIR = PROJECT_ROOT / "outputs" / "reports"

In [2]:
for file in REPORTS_DIR.glob("*tuned*"):
    print(file)

../outputs/reports/tuned_model_comparison.csv
../outputs/reports/final_baseline_vs_tuned.csv
../outputs/reports/tuned_classification_report.txt
../outputs/reports/tuned_final_metrics.csv


In [3]:
possible_tuning_files = [
    REPORTS_DIR / "tuning_results.csv",
    REPORTS_DIR / "tuned_results.csv",
    REPORTS_DIR / "best_params.csv"
]

for file in possible_tuning_files:
    print(file, file.exists())

../outputs/reports/tuning_results.csv False
../outputs/reports/tuned_results.csv False
../outputs/reports/best_params.csv False


In [4]:
tuning_results = pd.DataFrame({
    "model": [
        "xgboost_tuned",
        "extra_trees_tuned",
        "random_forest_tuned",
        "svm_rbf_tuned"
    ],
    "best_cv_f1_macro": [
        0.771417605004183,
        0.7622232799868008,
        0.7536844498162176,
        0.7502782054851086
    ],
    "best_params": [
        "{'colsample_bytree': 0.9, 'learning_rate': 0.1, 'max_depth': 4, 'n_estimators': 100, 'subsample': 0.8}",
        "{'max_depth': None, 'max_features': 'sqrt', 'min_samples_leaf': 2, 'n_estimators': 300}",
        "{'max_depth': None, 'max_features': 'sqrt', 'min_samples_leaf': 1, 'n_estimators': 700}",
        "{'model__C': 50, 'model__gamma': 0.001}"
    ]
})

display(tuning_results)

,model,best_cv_f1_macro,best_params
0,xgboost_tuned,0.771418,"{'colsample_bytree': 0.9, 'learning_rate': 0.1..."
1,extra_trees_tuned,0.762223,"{'max_depth': None, 'max_features': 'sqrt', 'm..."
2,random_forest_tuned,0.753684,"{'max_depth': None, 'max_features': 'sqrt', 'm..."
3,svm_rbf_tuned,0.750278,"{'model__C': 50, 'model__gamma': 0.001}"


## Tuning Summary

Among the tuned models, XGBoost obtained the highest cross-validation macro F1-score of 0.7714. The selected parameters were colsample_bytree = 0.9, learning_rate = 0.1, max_depth = 4, n_estimators = 100, and subsample = 0.8.